CPP/KMP/KMP.cpp

```diff
@@ -0,0 +1,74 @@
+// Program to implement the KMP algorithm in C++
+//Md Zeeshan
+

```

Comment:
Explain more about the problem, with some test cases and validate those test cases with your drive code.

Reply:
Bhaiya I added Version 2 of Circular Linked List. Please check it and command me if any changes is needed.

Reply:
Explanation needs to be present in the code file. 

Let's keep it simple. 
Think of a person who is just starting with coding, and comes across this repository and finds many programs across multiple languages. Now how that person is supposed to know what is the actual functionality of the code he is currently viewing if it find only a code block. 
Seems logical?

Have a look at [this](https://github.com/ShubhKotnala/Data_Structures_And_Algorithms/blob/main/java/array/MountainArray.java) file, and see how it explains its reader what the program is going to before actually starting with the code.